### DISCUSSION TITLE
Is there a way to delete duplicate rows? I

### DISCUSSION THREAD

**Discussion by jonathanmotif** [AntlerVC/firetable#873] [Support / Q&A] [answered]
Is there a way to delete duplicate rows? I imported a .csv and it created new rows instead of overwriting as I discovered. 

Now I have doubled my table and wish to roll back. 

Is there a. de-duplicate? b. Undo?

**Comment by harinij** [comment]
Hi @jonathanmotif unfortunately there isn't. While importing from CSV you have the option to select an ID column with which to make a match with and update instead of create new rows. There is also a preview of how your data would look like before the actual import. We will work on improving this process further with an undo option in the future.

**Comment by jonathanmotif** [comment]
Hi Harini,

Thanks for your reply.

I did match the import to a name column after reading another post you
responded to on GH. That didn't appear to work. I "practicised" this in a
dummy table beforehand.

Preview mode doesn't actually show duplicates in this instance because it
randomly samples row data. It does so in such a way so you cannot see them
side by side. T

What do you advise me to do next?

**Comment by htuerker** [comment, marked as answer]
Hey @jonathanmotif,

Unfortunately, there isn't a feature to revert the changes for now.

From my understanding, you must have imported the CSV data by auto-generated IDs at the first import which is the default option. Then, you expected by selecting a column as an ID would match your actual dataset at the latest import.

Selecting a column as an identifier just basically assigns the document ID with the CSV data as shown below. And it works properly only if your table data was imported in the same way. 

![Screen Shot 2022-10-19 at 00 34 48](https://user-images.githubusercontent.com/46192266/196549346-5c9efea4-b1a9-4702-b575-e0f65232e060.png)
![Screen Shot 2022-10-19 at 00 21 55](https://user-images.githubusercontent.com/46192266/196549626-641ce2cb-e3b5-403b-9e5b-bea73440102e.png)

You can also check from your [firebase console](https://console.firebase.google.com), the documents must be like shown below as their ids are identical to that column value.
 
![Screen Shot 2022-10-19 at 00 38 19](https://user-images.githubusercontent.com/46192266/196549815-1cd35f45-3041-4660-a10d-9c8d60c3899f.png)

So, it's strictly required to use the same ID format to match and update documents while importing. 

Finally, to remove those duplicates I'd suggest you run a script that reads the entire collection, respectively finds the documents created with that column data as an identifier, and deletes it.
